langchain/src/vectorstores/similarity-score-threshold.ts

```diff
@@ -0,0 +1,72 @@
+import { VectorStore, VectorStoreRetriever } from "./base.js";
+import { Document } from "../document.js";
+
+export type SimilarityScoreFilters = {
+  minSimilarityScore: number;
+  dynamicK?: boolean;
+  kIncrement?: number;
+  maxK?: number;
+};
+
+export class SimilarityScoreThresholdVectorStoreRetriever {
+  minSimilarityScore: number;
+
+  dynamicK = false;
+
+  kIncrement = 1;
+
+  maxK = 100;
+
+  constructor(similarityScoreFilters: SimilarityScoreFilters) {
+    this.minSimilarityScore = similarityScoreFilters.minSimilarityScore;
+    this.dynamicK = similarityScoreFilters.dynamicK ?? this.dynamicK;
+    this.kIncrement = similarityScoreFilters.kIncrement ?? this.kIncrement;
+    this.maxK = similarityScoreFilters.maxK ?? this.maxK;
+  }
+
+  private isAboveMinScore([, score]: [Document, number]): boolean {
+    return score >= this.minSimilarityScore;
+  }
+
+  fromVectorStore(vectorStoreRetriever: VectorStoreRetriever<VectorStore>) {
+    const isAboveMinScore = this.isAboveMinScore.bind(this);
+
+    // eslint-disable-next-line no-param-reassign
+    vectorStoreRetriever.getRelevantDocuments = async (query: string) => {
```

Comment:
Overriding a method like this can be dangerous - your retriever should instead extend `VectorStoreRetriever` and override the `getRelevantDocuments` and `addDocuments` methods. The constructor should take a `vectorStoreRetriever` and call that method in the overridden methods.

The issue with your current approach is potential side effects. If someone were to try to use the `vectorStoreRetriever` after calling this method, this updated implementation would be used rather than the expected one.

LMK if that doesn't make sense and I can help rework this.

Reply:
I understand the issue this can cause, but I must admit I don't quite understand how to fix it... Do you have an example of the code you're expecting?